demos/interactive_face_detection_demo/cpp/detectors.cpp

```diff
@@ -16,169 +15,102 @@
 #include <iterator>
 #include <map>
 
-#include <inference_engine.hpp>
-#include <ngraph/ngraph.hpp>
-
 #include <utils/ocv_common.hpp>
 #include <utils/slog.hpp>
 
 #include "detectors.hpp"
 
-BaseDetection::BaseDetection(const std::string &topoName,
-                             const std::string &pathToModel,
-                             const std::string &deviceForInference,
-                             int maxBatch, bool isBatchDynamic, bool isAsync,
-                             bool doRawOutputMessages)
-    : topoName(topoName), pathToModel(pathToModel), deviceForInference(deviceForInference),
-      maxBatch(maxBatch), isBatchDynamic(isBatchDynamic), isAsync(isAsync),
-      enablingChecked(false), _enabled(false), doRawOutputMessages(doRawOutputMessages) {
-    if (isAsync) {
-        slog::debug << "Use async mode for " << topoName << slog::endl;
-    }
-}
-
-BaseDetection::~BaseDetection() {}
-
-InferenceEngine::ExecutableNetwork* BaseDetection::operator ->() {
-    return &net;
-}
-
-void BaseDetection::submitRequest() {
-    if (!enabled() || request == nullptr) return;
-    if (isAsync) {
-        request->StartAsync();
-    } else {
-        request->Infer();
-    }
-}
+namespace {
+constexpr size_t ndetections = 200;
```

Comment:
It is replacement for maxProposalCount, isn't it? Then it was determined based on model shape, why now it's hardcoded?

Reply:
None of the plugins currently support https://github.com/openvinotoolkit/openvino/blob/069b3839cec79d26eb4bf8ee90a3b7437f96c106/src/inference/include/openvino/runtime/infer_request.hpp#L173
To work this around I create a batch of size equal to max possible number of detections. So non detection models need to know this limit, which is `ndetections`. And a detection model verifies that `ndetections = 200` is correct guess. After all detections are filed in a batch for non detection models, I shrink the tensor to be of smaller size to avoid inferencing all `ndetections` elements.

Reply:
Understand, thanks for explanation